src/app/dataset-view/additional-components/metadata-component/metadata.component.ts

```diff
@@ -117,7 +122,19 @@ export class MetadataComponent extends BaseComponent implements OnInit {
         if (this.currentState) {
             return (
                 this.datasetBasics.kind === DatasetKind.Root &&
-                !_.isNil(this.currentState.metadataSummary.metadata.currentSource) &&
+                !_.isNil(this.currentState.metadataSummary.metadata.currentPollingSource) &&
+                this.datasetPermissions.permissions.canCommit
+            );
+        } else {
+            return false;
+        }
+    }
+
+    public get canEditAddPushSource(): boolean {
+        if (this.currentState) {
+            return (
+                this.datasetBasics.kind === DatasetKind.Root &&
+                !!this.currentState.metadataSummary.metadata.currentPushSources.length &&
```

Comment:
I really don't like this TS syntax with implicit boolean conversions. Just check that length is more than zero, it's explicit and does not kill your brain.

Reply:
Fixed